crates/autopilot/src/run_loop.rs

```diff
@@ -74,14 +79,81 @@ impl RunLoop {
 
         // Shuffle so that sorting randomly splits ties.
         solutions.shuffle(&mut rand::thread_rng());
-        solutions.sort_unstable_by(|left, right| left.1.score.total_cmp(&right.1.score));
+        solutions.sort_unstable_by_key(|solution| solution.1.score);
 
         // TODO: Keep going with other solutions until some deadline.
         if let Some((index, solution)) = solutions.pop() {
             // The winner has score 0 so all solutions are empty.
-            if solution.score == 0. {
+            if solution.score == 0.into() {
                 return;
             }
+
+            let auction_id = id;
+            let winner = solution.reward.performance_address;
```

Comment:
I think this confuses reward payout address with the winning solver address which don't have to be the same.

Reply:
Makes sense. Are you maybe aware how we get the submission address for solvers under colocation?

Reply:
Same as here: https://github.com/cowprotocol/services/pull/1425#discussion_r1198807045